### PR TITLE
fixed dependencies processing in case of helm install or upgrade for …

### LINKED ISSUE #7497: Warning: ImportValues missing table: "dev" is not a table
Output of `helm version`:
version.BuildInfo{Version:"v3.0.3", GitCommit:"ac925eb7279f4a6955df663a0128044a8a6b7593", GitTreeState:"clean", GoVersion:"go1.13.6"}
Output of `kubectl version`:
Client Version: version.Info{Major:"1", Minor:"17", GitVersion:"v1.17.2", GitCommit:"59603c6e503c87169aea6106f57b9f242f64df89", GitTreeState:"clean", BuildDate:"2020-01-23T14:21:54Z", GoVersion:"go1.13.6", Compiler:"gc", Platform:"darwin/amd64"}
Server Version: version.Info{Major:"1", Minor:"14", GitVersion:"v1.14.10+IKS", GitCommit:"18e8565daf60eb3a20c0ac29a7d3a93622659e4d", GitTreeState:"clean", BuildDate:"2020-01-16T04:08:27Z", GoVersion:"go1.12.12", Compiler:"gc", Platform:"linux/amd64"}
Cloud Provider/Platform (AKS, GKE, Minikube etc.): 
IBM Kubernetes Service (IKS)

Hi guys, I have the following problem with the 'helm package' and the 'helm upgrade', I don't know if I'm using it wrong somehow, or if it's a bug.

After `helm package` the directory has the following structure
`helm package --version v1.0.1 --app-version v1.0.1 --dependency-update ./interaction-mgr`

```
tree
.
├── Chart.lock
├── Chart.yaml
├── charts
│   ├── dev-v0.1.0.tgz
│   └── prod-v0.1.0.tgz
├── envs
│   ├── dev
│   │   ├── Chart.yaml
│   │   └── values.yaml
│   └── prod
│       ├── Chart.yaml
│       └── values.yaml
├── templates
│   ├── NOTES.txt
│   ├── _helpers.tpl
│   ├── autoscaler.yaml
│   ├── deployment.yaml
│   ├── istio-destination-rule.yaml
│   ├── istio-policy.yaml
│   ├── istio-virtualservice.yaml
│   └── service.yaml
└── values.yaml
```

Content of parent Chart.yaml
```
apiVersion: v2
name: interaction-mgr
version: v0.1.0
appVersion: v0.1.0
dependencies:
  - name: dev
    repository: "file://envs/dev"
    version: ">= 0.0.1"
    tags:
      - dev
    import-values:
      - data

  - name: prod
    repository: "file://envs/prod"
    version: ">= 0.0.1"
    tags:
      - prod
    import-values:
      - data
```

Content of envs/dev/Chart.yaml
```
apiVersion: v2
name: dev
version: v0.1.0
appVersion: v0.1.0
```
Content of envs/dev/values.yaml
```
nameOverride: interaction-mgr
exports:
  data:
    replicaCount: 1
    image:
      pullPolicy: Always
    telemetry:
      level: "Information"
    reason: "new dev deploy"
    current: true
    resources:
      requests:
        cpu: 100m
        memory: 256Mi
      limits:
        cpu: 500m
        memory: 500Mi
      autoscaler:
        minReplicas: 1
        maxReplicas: 3
        targetCPUUtilizationPercentage: 90
    cors:
      allowOrigin:
      # - test.com
```

Content of envs/prod/Chart.yaml
```
apiVersion: v2
name: prod
version: v0.1.0
appVersion: v0.1.0
```
Content of envs/prod/values.yaml
```
nameOverride: interaction-mgr
exports:
  data:
    replicaCount: 3
    image:
      pullPolicy: Always
    telemetry:
      level: "Error"
    reason: "new prod deploy"
    current: true
    resources:
      requests:
        cpu: 350m
        memory: 350Mi
      limits:
        cpu: 500m
        memory: 500Mi
      autoscaler:
        minReplicas: 2
        maxReplicas: 5
        targetCPUUtilizationPercentage: 90
    cors:
      allowOrigin:
      # - test.com
```

When I now perform `helm upgrade` command, I always receive the following error message and my Azure DevOps Pipeline fails. However, the command is successful and the new release is installed with the `prod` values.

`helm upgrade --install  --set image.tag=v1.0.1,tags.prod=true --version v1.0.1 interaction-mgr-v1 interaction-mgr-v1.0.1.tgz`

Error message: 
`dependencies.go:249: Warning: ImportValues missing table: "dev" is not a table`

I already investigated some time in the source code.
https://github.com/helm/helm/blob/45d986327a544689d9199b2d71183d7d9f21e696/pkg/chartutil/dependencies.go#L248-L250
Maybe this issue could be fixed with:
```
				if err != nil && r.Enabled {
					log.Printf("Warning: ImportValues missing table: %v", err)
					continue
				}
```

### PR DESCRIPTION
**Pull request by fhopfensperger**
fixes #7497 
I have noticed that `helm install/upgrade` tries to resolve all dependencies. In the case that individual sub-charts are disabled, the function `processDependencyEnabled` will delete disabled sub-charts. But the deactivated charts remain in the chart metadata. In the next step, the function `processImportValues` tries to resolve all sub-charts which are defined in the metadata. Since the dependency has already been deleted, the `Table` function tries to grab the no longer available sub-chart and throws the described error.

Signed-off-by: Florian Hopfensperger <florian.hopfensperger@de.ibm.com>

Note: This is my first open-source contribution, it may well be that I made a mistake.

### CODE REVIEW

**Review by hickeyma — APPROVED** [review]
LGTM, thanks for the fix @fhopfensperger

**Comment by hickeyma** [comment]
@fhopfensperger Before I merge, it needs another review from @helm/helm-core-maintainers as the size is large.

**Review by technosophos — APPROVED** [review]
Code LGTM

**Pull request merged**
